### ISSUE TITLE
Mailer#toNotify's Future is likely to be executed before the transaction is committed.

### ISSUE THREAD

**Issue by troter** [takezoe/gitbucket#689]
When create issue or open new pull request, gitbucket will try to send a notification.
But currently, It may not be send notification. Because Servlet Filter's database session and Mailer's database session are different, and Mailer#toNotify's Future is likely to be executed before the transaction is committed.

This is a dirty patch.
```diff
diff --git a/src/main/scala/gitbucket/core/util/Notifier.scala b/src/main/scala/gitbucket/core/util/Notifier.scala
index 9e80917..27fcf9f 100644
--- a/src/main/scala/gitbucket/core/util/Notifier.scala
+++ b/src/main/scala/gitbucket/core/util/Notifier.scala
@@ -73,6 +73,7 @@ class Mailer(private val smtp: Smtp) extends Notifier {

     val f = Future {
       database withSession { implicit session =>
+        Thread.sleep(10*1000)
         getIssue(r.owner, r.name, issueId.toString) foreach { issue =>
           defining(
               s"[${r.name}] ${issue.title} (#${issueId})" ->
```

**Comment by takezoe** [comment]
@shimamoto Do you have any idea to solve this issue?

**Comment by rlazoti** [comment]
@takezoe I've already fixed it, I'm going to send you a PR. :)

**Comment by takezoe** [comment]
Fixed in #699
